src/test/java/org/zalando/nakadi/service/timeline/TimelineServiceTest.java

```diff
@@ -0,0 +1,180 @@
+        Mockito.when(topicRepository.createTopic(Matchers.anyInt(), Matchers.anyLong())).thenReturn("topic");
+        Mockito.when(eventTypeCache.getEventType(Matchers.any())).thenReturn(EventTypeTestBuilder.builder().build());
+
+        timelineService.createTimeline("event_type", "storage_id", new FullAccessClient("clientId"));
+
+        Mockito.verify(timelineDbRepository).createTimeline(Matchers.any());
+        Mockito.verify(timelineSync)
+                .startTimelineUpdate(Matchers.eq(activeTimeline.getEventType()), Matchers.anyLong());
+        Mockito.verify(timelineDbRepository).updateTimelime(Matchers.eq(activeTimeline));
+        Mockito.verify(timelineDbRepository).updateTimelime(Matchers.eq(activeTimeline));
+        Mockito.verify(timelineSync).finishTimelineUpdate(activeTimeline.getEventType());
+    }
+
+    @Test
+    public void testDeleteTimeline() throws Exception {
+        final EventType eventType = EventTypeTestBuilder.builder().build();
+        final String timelineId = "a677373f-7fc2-42bd-ab62-3d52781dcd9d";
+        final Timeline timeline = Timeline.createTimeline(eventType.getName(), 0, null, "topic", new Date());
+        timeline.setId(UUID.fromString(timelineId));
+        final ImmutableList<Timeline> timelines = ImmutableList.of(timeline);
+
+
+        Mockito.when(securitySettings.getAdminClientId()).thenReturn("clientId");
+        Mockito.when(eventTypeCache.getEventType(Matchers.any())).thenReturn(eventType);
+        Mockito.when(timelineDbRepository.listTimelines(eventType.getName())).thenReturn(timelines);
+
+        timelineService.delete(eventType.getName(), timelineId, new FullAccessClient("clientId"));
+
+        Mockito.verify(timelineSync).startTimelineUpdate(Matchers.eq(eventType.getName()), Matchers.anyLong());
+        Mockito.verify(timelineDbRepository).deleteTimeline(UUID.fromString(timelineId));
+        Mockito.verify(timelineSync).finishTimelineUpdate(eventType.getName());
+    }
+
+    @Test
+    public void testGetTimelines() throws Exception {
+        final EventType eventType = EventTypeTestBuilder.builder().build();
+        final ImmutableList<Timeline> timelines = ImmutableList.of(
+                Timeline.createTimeline("event_type", 0, null, "topic", new Date()),
+                Timeline.createTimeline("event_type_1", 1, null, "topic_1", new Date()));
+
+        Mockito.when(securitySettings.getAdminClientId()).thenReturn("clientId");
+        Mockito.when(eventTypeCache.getEventType(Matchers.any())).thenReturn(eventType);
+        Mockito.when(timelineDbRepository.listTimelines(eventType.getName())).thenReturn(timelines);
+
+        final List<Timeline> actualTimelines =
+                timelineService.getTimelines(eventType.getName(), new FullAccessClient("clientId"));
+        Assert.assertEquals(timelines, actualTimelines);
+    }
+
+    @Test(expected = NotFoundException.class)
+    public void testGetTimelinesNotFound() throws Exception {
```

Comment:
This one is fine. I think that main scenarios like create from fake and create real could be done only in acceptance tests. Then, corner case scenarios could be exercised here through unit tests with small number of mocks.